toolkit/scripts/update_kernel.sh

```diff
@@ -0,0 +1,275 @@
+    sed -i "s/.*$FILE_SIGNATURE_PATTERN.*/$FULL_SHA256SUM_ENTRY/" $SHA256SUM_FILE
+}
+
+function update_toolchain_scripts {
+    TOOLCHAIN_FOLDER=$WORKSPACE/toolkit/scripts/toolchain/
+    TOOLCHAIN_SCRIPTS="toolchain_build_in_chroot.sh toolchain_build_temp_tools.sh"
+    for script in $TOOLCHAIN_SCRIPTS
+    do
+        file=$TOOLCHAIN_FOLDER/container/$script
+        PATTERN="echo Linux-.* API Headers"
+        REPLACE="echo Linux-$VERSION API Headers"
+        sed -i "s/$PATTERN/$REPLACE/" $file
+        PATTERN="tar xf $FILE_SIGNATURE_PATTERN.*.tar.gz"
+        REPLACE="tar xf $TARBALL_NAME"
+        sed -i "s/$PATTERN/$REPLACE/" $file
+        PATTERN="CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-.*"
+        REPLACE="CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-$VERSION"
+        sed -i "s/$PATTERN/$REPLACE/" $file
+    done
+}
+
+function update_toolchain_wget_url {
+    FILE=$WORKSPACE/toolkit/scripts/toolchain/container/toolchain-remote-wget-list
+    PATTERN="$DEFAULT_URL.*"
+    REPLACE="$FULL_URL"
+    sed -i "s#$PATTERN#$REPLACE#" $FILE
+}
+
+function update_toolchain_dockerfile {
+    FILE=$WORKSPACE/toolkit/scripts/toolchain/container/Dockerfile
+    PATTERN="$DEFAULT_URL.* -O"
+    REPLACE="$FULL_URL -O"
+    sed -i "s#$PATTERN#$REPLACE#" $FILE
+    PATTERN="kernel-.*.tar.gz"
+    REPLACE="$TARBALL_NAME"
+    sed -i "s#$PATTERN#$REPLACE#" $FILE
+}
+
+function update_toolchain_pkglist {
+    PKGLIST_FOLDER=$WORKSPACE/toolkit/resources/manifests/package/
+    PKGLIST="pkggen_core_aarch64.txt pkggen_core_x86_64.txt toolchain_aarch64.txt toolchain_x86_64.txt"
+    for pkg in $PKGLIST
+    do
+        file=$PKGLIST_FOLDER/$pkg
+        PATTERN="kernel-headers-.*"
+        REPLACE="kernel-headers-$VERSION-1.cm2.noarch.rpm"
+        sed -i "s/$PATTERN/$REPLACE/" $file
+    done
+}
+
+function update_toolchain {
+    #update_toolchain_md5sum
+    update_toolchain_sha256sum
+    update_toolchain_scripts
+    update_toolchain_pkglist
+    update_toolchain_dockerfile
+}
+
+function replace_cgversion {
+    for spec in $SPECS
+    do
+        PATTERN="\"name\": \"$spec\","
+        REPLACE="\ \ \ \ \ \ \ \ \ \ \"version\": \"$VERSION\","
+        sed -i "/$PATTERN/!b;n;c$REPLACE" $1
+    done
+}
+
+function update_cgmanifest {
+    CGMANIFEST_FILE=$WORKSPACE/cgmanifest.json
+    # Replace URL
+    PATTERN="$DEFAULT_URL.*"
+    REPLACE="$FULL_URL\""
+    sed -i "s#$PATTERN#$REPLACE#" $CGMANIFEST_FILE
+    # Replace version
+    replace_cgversion $CGMANIFEST_FILE
+}
+
+function print_metadata {
+    MD5="$(md5sum $DOWNLOAD_FILE_PATH | awk '{print $1;}')"
+    SHA256="$(sha256sum $DOWNLOAD_FILE_PATH | awk '{print $1;}')"
+    echo md5sum = $MD5
+    echo sha256 = $SHA256
+}
+
+
+function usage() { 
+    echo "Update sources for kernel" 
+    echo "v : Version you are updating to (ex. 5.10.37.1)" 
+    echo "u : Your name"
+    echo "e : Your email"
+    echo "w : Absoulte patch to your workspace for your update - no quotes"
```

Comment:
Did you mean path?

Reply:
nit - Also `s/Absoulte/Absolute`
```suggestion
    echo "w : Absolute path to your workspace for your update - no quotes"
```